File: day seven/menstrualWahala/MenstrualCycleCalculator.java
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.util.Scanner;

public class MenstrualCycleCalculator{

	public static void main(String[] args){

		Scanner scanner = new Scanner(System.in);
	
		System.out.println("WELCOME TO THE MENSTRUAL CYCLE CALCULATOR");

		System.out.println();
	
		System.out.println("Assumed Length Of Cycle Is Between 21 - 35 Days");

		System.out.println();
		
		System.out.print("Enter The Date Of Your Last Menstrual Flow (DD-MM-YYYY): ");	

		String lastFlowDate = scanner.nextLine();

		System.out.print("Enter The Average Length Of Your Cycle In Days: ");

		int cycleLength = scanner.nextInt();
		
		DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");

		LocalDate lastFlowDatePeriod = LocalDate.parse(lastFlowDate,formatter);
		
		LocalDate nextMenstrualFlow = lastFlowDatePeriod.plusDays(cycleLength);

		System.out.print("Your Next Menstrual Flow Starts On: " + nextMenstrualFlow);

		LocalDate ovulationDate = nextMenstrualFlow.minusDays(14);

		System.out.print("\nYour Ovulation Date Is: " + ovulationDate + "\n");

		LocalDate fertilePeriodStart = ovulationDate.minusDays(7);

        	LocalDate fertilePeriodEnd = ovulationDate.plusDays(2);

		System.out.print("Your Fertile Period Is Between : " + ovulationDate.minusDays(7) + " - " + ovulationDate.plusDays(2) +"\n");

		LocalDate safePeriodStart = ovulationDate.plusDays(5);

		LocalDate safePeriodEnd = nextMenstrualFlow.minusDays(5);

		System.out.println("Your Safe Period Is Between: " + ovulationDate.plusDays(5) + " - " + nextMenstrualFlow.minusDays(5) );


	}

}

File: day seven/menstrualWahala/Menstrualcyclecalculator.js
const prompt = require("prompt-sync")();

console.log("WELCOME TO THE MENSTRUAL CYCLE CALCULATOR");

console.log();

console.log("Assumed Length Of Cycle Is Between 21 - 35 Days");

console.log();

const lastFlowDate = prompt("Enter The Date Of Your Last Menstrual Flow (DD-MM-YYYY): ");

const cycleLength = parseInt(prompt("Enter The Average Length Of Your Cycle In Days: "));

const lastFlowDatePeriod = new Date(lastFlowDate);

const nextMenstrualFlow = new Date(lastFlowDatePeriod.getTime() + cycleLength );

console.log("Your Next Menstrual Flow Starts On: " + nextMenstrualFlow.toLocaleDateString());

const ovulationDate = new Date(nextMenstrualFlow.getTime() - 14 );

console.log("\nYour Ovulation Date Is: " + ovulationDate.toLocaleDateString() + "\n");

const fertilePeriodStart = new Date(ovulationDate.getTime() - 7 );

const fertilePeriodEnd = new Date(ovulationDate.getTime() + 2 );

console.log("Your Fertile Period Is Between : " + fertilePeriodStart.toLocaleDateString() + " - " + fertilePeriodEnd.toLocaleDateString() + "\n");

const safePeriodStart = new Date(ovulationDate.getTime() + 5 );

const safePeriodEnd = new Date(nextMenstrualFlow.getTime() - 5 );

console.log("Your Safe Period Is Between: " + safePeriodStart.toLocaleDateString() + " - " + safePeriodEnd.toLocaleDateString());

